### PR TITLE
Simplified configuration of samples for NCS.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem

prj.conf files in sample applications for nrfconnect use some obsolete
config options, which are set by default. Moreover many config options will be
duplicated for every NCS sample.

#### Summary of Changes

- Removed options, which are set by default,

- Created overlay-sample-defaults.conf file storing configs shared by all samples,

- Added Kconfig option for enabling using CHIP,

- Increased OpenThread stack size to 6 kB to avoid stack overflow during Thread joining.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: config/nrfconnect/overlay-sample-defaults.conf [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#
+#   Copyright (c) 2020 Nordic Semiconductor
```
Comment (BroderickCarlin) [inline config/nrfconnect/overlay-sample-defaults.conf:2]: Should this file have the CHIP copyright?

Comment (rwalker-apple) [reply config/nrfconnect/overlay-sample-defaults.conf:2]: given where it lives, I think so...

**Review by rwalker-apple — APPROVED** [review]

File: config/nrfconnect/overlay-sample-defaults.conf (hunk 1 above)
Comment (kkasperczyk-no) [reply config/nrfconnect/overlay-sample-defaults.conf]: Ok, it's done.

**Review by LuDuda — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: config/nrfconnect/overlay-sample-defaults.conf (hunk 1 above)
Comment (kkasperczyk-no) [reply config/nrfconnect/overlay-sample-defaults.conf]: @BroderickCarlin could you verify if you are ok with this now and approve or request further changes?

**Comment by mspang** [comment]
@project-chip/reviewers-amazon @project-chip/reviewers-comcast @project-chip/reviewers-samsung

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**